File: algorithm/typescript/BOJ3052.ts
// problem : https://www.acmicpc.net/problem/3052

import fs = require("fs");

const input: number[] = fs.readFileSync("/dev/stdin").toString().trim().split("\n").map(Number);

const set = new Set();
input.forEach((num) => {
  set.add(num % 42);
});

console.log(set.size);

File: algorithm/java/src/BOJ3052.java
// problem : https://www.acmicpc.net/problem/3052

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.HashSet;

public class BOJ3052 {
    static final int SIZE = 10;
    static HashSet<Integer> hashSet = new HashSet<>();

    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        for (int i = 0; i < SIZE; i++) {
            int num = Integer.parseInt(br.readLine());
            hashSet.add(num % 42);
        }

        System.out.println(hashSet.size());
    }
}
